src/templates/admin/elements/fundref/fundref.html

```diff
@@ -1,184 +1,76 @@
-
-                                    const add_funder_header = document.createElement('h5');
-                                    add_funder_header.innerText = '{% trans "Add funder" %}';
-                                    card_divider_div.appendChild(add_funder_header);
-
-                                    const add_funder_card_section = document.createElement('div');
-                                    add_funder_card_section.classList.add('card-section');
-                                    card_div.appendChild(add_funder_card_section);
-
-                                    const add_funder_form = document.createElement('form');
-                                    add_funder_form.setAttribute('method', 'POST');
-                                    add_funder_card_section.appendChild(add_funder_form);
-                                    add_funder_form.innerHTML = "<input type=\"hidden\" name=\"csrfmiddlewaretoken\" value=\"{{csrf_token}}\">"
+{% load i18n foundation %}
 
+<!-- Note: this is transcluded as templated HTML because we need to pass: 1. translation templates; and 2. CSRF tokens-->
+<script>
+    function search_fundref(query) {
+        query_url = '//api.crossref.org/funders?query=' + query;
+
+        // Clear out old modals
+        console.log('Clearing old modals.')
+        $("div.fundref_modal").remove();
+        console.log('Fetching ' + query_url);
+        $("#funder_list").html('<p class="text-center"><i class="fa fa-circle-o-notch fa-spin"></i><br />Searching Fundref...</p>');
+        fetch(query_url)
+            .then(response => response.json())
+            .then(json => {
+                const funder_span = $('#funder_list');
+                funder_span.empty();
+
+                // Prep the table for display
+                const table = $('<table>').addClass('scroll');
+                const thead = $('<thead>').appendTo(table);
+                const thead_tr = $('<tr>').appendTo(thead);
+                $('<th>').attr('colspan', '2').text('{% trans "Funder" %}').appendTo(thead_tr);
+
+                // Remove TinyMCE for the cloning process.
+                tinymce.remove();
+
+                // Loop across each funder in the fundref response, add them to
+                // the table and create a modal for them.
+                $.each(json.message.items, (index, funder) => {
+                    const funder_tr = $('<tr>').appendTo(table);
+                    $('<td>').text(funder.name).appendTo(funder_tr);
+                    const add_funder_td = $('<td>').appendTo(funder_tr);
+                    const add_funder_button = $('<a>')
+                        .addClass('success button pull-right')
+                        .attr({
+                            'href': '#',
+                            'data-open': 'add_funder_' + index,
+                            'name': 'add_funder_' + index
+                        })
+                        .appendTo(add_funder_td);
```

Comment:
This button is not working for me in the Edit Metadata pane. It works in submission.

This might be related to the `modal_clones_placeholder` only existing in submission, but I also inspected this in the browser and nothing gets added to that div, even when the feature works. They go to the end of the body.

If you find you need to refactor, I think I have an idea about how to do it without creating lots of modals. Just attach a click event listener to each Add button that modifies the original modal with the details for the selected funder. Define the function that modifies the modal outside the for loop that creates the table, so that you can more clearly specify and control inputs like you would with a Python function. Might this also be a way to avoid having to reinitialize TinyMCE and Foundation?

Reply:
The next time my brain says "this should be an issue of its own" I'm going to listen. I've rewritten this where the loop stores the details in data attributes on the button. When the button is clicked it fills the details in the modal and pops it open.